sunpy/net/helioviewer.py

```diff
@@ -71,29 +91,42 @@ def get_closest_image(self, date, **kwargs):
         --------
         >>> from sunpy.net import helioviewer
         >>> client = helioviewer.HelioviewerClient()  # doctest: +REMOTE_DATA
-        >>> metadata = client.get_closest_image('2012/01/01', sourceId=11)  # doctest: +REMOTE_DATA
+        >>> metadata = client.get_closest_image('2012/01/01', sourceid=11)  # doctest: +REMOTE_DATA
         >>> print(metadata['date'])  # doctest: +REMOTE_DATA
         2012-01-01 00:00:07
         """
+
+        if('sourceid' in kwargs):
+            source_id = kwargs['sourceid']
+        else:
+            key = (observatory, instrument, detector, measurement)
+            source_id = self.datasource_info[key]
+
         params = {
             "action": "getClosestImage",
-            "date": self._format_date(date)
+            "date": self._format_date(date),
+            "sourceId": source_id,
+            "observatory": observatory,
+            "instrument": instrument,
+            "detector": detector,
+            "measurement": measurement
         }
         params.update(kwargs)
-
+        
```

Comment:
Could you remove the whitespace on this line?

Reply:
Ok sure

Reply:
I can't find those white space in my local file